File: 1. Two Sum.py
# https://leetcode.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        for i in range(len(nums)):
            if target - nums[i] in nums[i+1::]:
                return [i, nums[i + 1::].index(target - nums[i]) + i + 1]
        

File: Leetcode/1. Two Sum/1. Two Sum.js
var twoSum = function(nums, target) {
    let i = 0
    while (i < nums.length) {
        let j = i + 1
        while (j < nums.length) {
            if (nums[i] + nums[j] === target) {
                return [i, j]
            }
            j++
        }
        i++
    }
}



var twoSum = function(nums, target) {
    let xdict = {}
    for (let i = 0; i < nums.length; i++) {
        y = target - nums[i]
        if (y in xdict) return [xdict[y], i]
        xdict[nums[i]] = i
    }
}